app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java

```diff
@@ -94,9 +94,17 @@ protected void onStart() {
     public void onBackPressed() {
         if (binding.drawerLayout.isDrawerOpen(GravityCompat.START)) {
             binding.drawerLayout.closeDrawer(GravityCompat.START);
-        } else {
+        }
+        else if(lastSelectedNavItemId!=R.id.nav_dashboard) {
+            binding.navView.setCheckedItem(R.id.nav_dashboard);
+            lastSelectedNavItemId = R.id.nav_dashboard;
+            Fragment fragment = EventDashboardFragment.newInstance(eventId);
+            fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+        }
```

Comment:
This is not using fragment backstack at all, instead this is creating a new instance of DashBoardFragment which shouldn't be the case

Reply:
Yes I did this deliberately because if we use addtobackstack() method to add fragments to the backstack then what will happen is suppose if we go to faqs and then from faqs to tickets, then both will be added to the stack and then pressing back would lead to faqs and then again pressing back will lead to the dashboard. I firstly implemented the backstack, but to jump from any fragment to dashboard we need to loop through each item and pop from the stack.I'll implement it using backstack if you want me to.

Reply:
Thats easy to handle and standard. Everytime to add to backstack you just update a boolean field and check if we are already one level on backstack and only add it if we aren't same with the popping of backstack